pkg/plugins/manager/loader/angulardetector/angulardetector.go

```diff
@@ -40,43 +38,9 @@ func (d *regexDetector) Detect(moduleJs []byte) bool {
 	return d.regex.Match(moduleJs)
 }
 
-// angularDetectors contains all the detectors to detect Angular plugins.
-// They are executed in the specified order.
-var angularDetectors = []detector{
-	&containsBytesDetector{pattern: []byte("PanelCtrl")},
-	&containsBytesDetector{pattern: []byte("QueryCtrl")},
-	&containsBytesDetector{pattern: []byte("app/plugins/sdk")},
-	&containsBytesDetector{pattern: []byte("angular.isNumber(")},
-	&containsBytesDetector{pattern: []byte("editor.html")},
-	&containsBytesDetector{pattern: []byte("ctrl.annotation")},
-	&containsBytesDetector{pattern: []byte("getLegacyAngularInjector")},
-
-	&regexDetector{regex: regexp.MustCompile(`['"](app/core/utils/promiseToDigest)|(app/plugins/.*?)|(app/core/core_module)['"]`)},
-	&regexDetector{regex: regexp.MustCompile(`from\s+['"]grafana\/app\/`)},
-	&regexDetector{regex: regexp.MustCompile(`System\.register\(`)},
-}
-
-// Inspect open module.js and checks if the plugin is using Angular by matching against its source code.
-// It returns true if module.js matches against any of the detectors in angularDetectors.
-func Inspect(p *plugins.Plugin) (isAngular bool, err error) {
-	f, err := p.FS.Open("module.js")
-	if err != nil {
-		return false, fmt.Errorf("open module.js: %w", err)
-	}
-	defer func() {
-		if closeErr := f.Close(); closeErr != nil && err == nil {
-			err = fmt.Errorf("close module.js: %w", closeErr)
-		}
-	}()
-	b, err := io.ReadAll(f)
-	if err != nil {
-		return false, fmt.Errorf("module.js readall: %w", err)
-	}
-	for _, d := range angularDetectors {
-		if d.Detect(b) {
-			isAngular = true
-			break
-		}
-	}
-	return
+// Inspector can inspect a module.js and determine if it's an Angular plugin or not.
+type Inspector interface {
```

Comment:
This reminds me of the existing interface we use to permit unsigned plugins - https://github.com/grafana/grafana/blob/main/pkg/plugins/ifaces.go#L128-L131. It's a shame that it's currently only used inside the signature validation, but maybe this can be reused some way generically?

Reply:
Agree, we should try unify these things. Doesn't have to happen in this PR though,  I guess.

Reply:
Or wait, you are using this information to warn in the UI, right? If so, please disregard.

Reply:
good point, but it may be tricky to merge the signature validation and this angular validator. Probably worth it to evaluate if it makes sense (or looks good) in a different PR

Reply:
Thanks for the feedback, I'll keep it in mind for a follow-up PR :smile: